src/entity/RDFEntity.cpp

```diff
@@ -1,81 +1,87 @@
 #include <sempr/entity/RDFEntity.hpp>
 #include <RDFEntity_odb.h>
-#include <Soprano/Soprano>
 
 namespace sempr { namespace entity {
 
-SEMPR_ENTITY_SOURCE(RDFEntity)
 
-RDFEntity::RDFEntity(const core::IDGenBase* idgen)
-    : Entity(idgen)
+// default implementation of the TripleIteratorWrapper: Does not return a triple, and every iterator equals another!
+TripleIteratorWrapper::TripleIteratorWrapper(TripleIterator* impl)
+    : impl_(impl)
 {
-    setDiscriminator<RDFEntity>();
 }
 
-
-RDFEntity::RDFEntity() : RDFEntity(new core::IDGen<RDFEntity>())
+TripleIteratorWrapper::~TripleIteratorWrapper()
 {
+    delete impl_;
 }
 
-std::vector<Triple>::const_iterator RDFEntity::begin() const
+const Triple TripleIteratorWrapper::operator*() const
 {
-    return triples_.begin();
+    if (!impl_)
+        throw std::exception();
```

Comment:
Didn't you trust yourself?
The `impl_` point could be stored as const in the wrapper and the c'tor can check the null expr.
These functions could also be declared as inline (in a future release).

Reply:
Well... the TripleIteratorWrapper relies on the user providing him with a valid implementation. I don't know, it just felt right to check if I actually set it. Though this is probably a bit exaggerated... What do you want me to change? Just get rid of it? Or leave it as it is? Or do the check in the c'tor to fail more early?

Reply:
Like this? https://github.com/sempr-tk/sempr/pull/37/commits/4a6d8e4379fd7291b56a0bf3b1aa07df7f638c00